ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -112,73 +119,21 @@ private void handleThrottlingInHeader(SdkHttpFullResponse sdkResponse) {
-            log.error("Remote server returned error code: {}", statusCode);
-            executionContext
-                .getExceptionHolder()
-                .compareAndSet(
-                    null,
-                    new OpenSearchStatusException(
-                        REMOTE_SERVICE_ERROR + "The request was denied due to remote server throttling.",
-                        RestStatus.fromCode(statusCode)
-                    )
-                );
-        }
-    }
-
-    private void processResponse(
-        Integer statusCode,
-        String body,
-        Map<String, String> parameters,
-        Map<Integer, ModelTensors> tensorOutputs
-    ) {
-        if (Strings.isBlank(body)) {
-            log.error("Remote model response body is empty!");
-            if (executionContext.getExceptionHolder().get() == null) {
-                executionContext
-                    .getExceptionHolder()
-                    .compareAndSet(null, new OpenSearchStatusException("No response from model", RestStatus.BAD_REQUEST));
-            }
-        } else {
-            if (statusCode < HttpStatus.SC_OK || statusCode > HttpStatus.SC_MULTIPLE_CHOICES) {
-                log.error("Remote server returned error code: {}", statusCode);
-                if (executionContext.getExceptionHolder().get() == null) {
-                    executionContext
-                        .getExceptionHolder()
-                        .compareAndSet(null, new OpenSearchStatusException(REMOTE_SERVICE_ERROR + body, RestStatus.fromCode(statusCode)));
-                }
-            } else {
-                try {
-                    ModelTensors tensors = processOutput(body, connector, scriptService, parameters, mlGuard);
-                    tensors.setStatusCode(statusCode);
-                    tensorOutputs.put(executionContext.getSequence(), tensors);
-                } catch (Exception e) {
-                    log.error("Failed to process response body: {}", body, e);
-                    if (executionContext.getExceptionHolder().get() == null) {
-                        executionContext
-                            .getExceptionHolder()
-                            .compareAndSet(null, new MLException("Fail to execute predict in aws connector", e));
-                    }
-                }
-            }
-        }
-    }
-
-    // Only all requests successful case will be processed here.
-    private void reOrderTensorResponses(Map<Integer, ModelTensors> tensorOutputs) {
-        ModelTensors[] modelTensors = new ModelTensors[tensorOutputs.size()];
-        log.debug("Reordered tensor outputs size is {}", tensorOutputs.size());
-        for (Map.Entry<Integer, ModelTensors> entry : tensorOutputs.entrySet()) {
-            modelTensors[entry.getKey()] = entry.getValue();
+        if (containsThrottlingException) {
+            handleException(
+                new SageMakerThrottlingException(
```

Comment:
The class name `SageMakerThrottlingException` is limited to Sagemaker, let's use a more general name

Reply:
How about RemoteConnectorThrottlingException?